rainier-core/src/main/scala/com/stripe/rainier/core/Injection.scala

```diff
@@ -17,21 +17,29 @@ trait Injection { self =>
     See https://en.wikipedia.org/wiki/Probability_density_function#Dependent_variables_and_change_of_variables
     This function should be log(d/dy backwards(y)), where y = forwards(x).
    */
-  def logJacobian(y: Real): Real
+  def logBackwardsJacobian(y: Real): Real
+  def logForwardsJacobian(x: Real): Real
 
   def transform(dist: Continuous): Continuous = new Continuous {
     def realLogDensity(real: Real): Real =
       If(isDefinedAt(real),
          dist.realLogDensity(backwards(real)) +
-           logJacobian(real),
+           logBackwardsJacobian(real),
          Real.zero.log)
 
     val generator: Generator[Double] =
       Generator.require(self.requirements) { (r, n) =>
         n.toDouble(forwards(dist.generator.get(r, n)))
       }
 
-    def param: RandomVariable[Real] = dist.param.map(forwards)
+    override def supportTransform(v: Variable): Real =
+      forwards(dist.supportTransform(v))
+
+    override def supportTransformLogJacobian(v: Variable): Real =
+      logForwardsJacobian(supportTransform(v)) + dist
+        .supportTransformLogJacobian(v)
+
+    override def param: RandomVariable[Real] = dist.param.map(forwards)
```

Comment:
This stuff gets confusing, but I believe you should just delete this and leave it to the default implementation, which will call `supportTransform` and `supportTransformLogJacobian`, otherwise those won't ever get called.

Reply:
I rewrote using `Support`, hopefully fixes this.